scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -0,0 +1,1117 @@
+            self.par = NULL
+        if self.rng != NULL:
+            unur_free(self.rng)
+            self.rng = NULL
+        if self.urng != NULL:
+            unur_urng_free(self.urng)
+            self.urng = NULL
+
+    # Pickling support
+    @cython.final
+    def __reduce__(self):
+        klass = functools.partial(self.__class__, **self._kwargs)
+        return (klass, ())
+
+
+cdef class TransformedDensityRejection(Method):
+    r"""
+    TransformedDensityRejection(dist, mode=None, center=None, *, params=(), domain=None, c=-0.5, cpoints=30, variant="ps", use_dars=True, max_sqhratio=0.99, max_intervals=100, use_center=True, use_mode=True, guide_factor=2, seed=None)
+
+    Transformed Density Rejection (TDR) Method.
+
+    TDR is an acceptance/rejection method that uses the concavity of a
+    transformed density to construct hat function and squeezes automatically.
+    Most universal algorithms are very slow compared to algorithms that are
+    specialized to that distribution. Algorithms that are fast have a slow
+    setup and require large tables. The aim of this universal method is to
+    provide an algorithm that is not too slow and needs only a short setup.
+    This method can be applied to univariate and unimodal continuous
+    distributions with T-concave density function. See [1]_ and [2]_ for
+    more details.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with 'pdf' and 'dpdf' methods.
+
+        * 'pdf': PDF of the distribution. The signature of the PDF is
+          expected to be: ``def pdf(x: float, *params) -> float``. i.e.
+          the PDF should accept a Python float and the parameters and
+          return a Python float. It doesn't need to integrate to 1 i.e.
+          the PDF doesn't need to be normalized.
+        * 'dpdf': Derivative of the PDF w.r.t x (i.e. the variate). Must
+          have the same signature as the PDF.
+
+    mode : float, optional
+        (Exact) Mode of the distribution.
+    center : float, optional
+        Approximate location of the mode or the mean of the distribution.
+        This location provides some information about the main part of the
+        PDF and is used to avoid numerical problems.
+    params : tuple, optional
+        Parameters that the PDF and DPDF take. Default is an empty tuple.
+    domain : list or tuple of length 2, optional
+        The support of the distribution. Must not contain `nan` values.
+        Default is ``None``. When ``None``, the support is assumed to be
+        :math:`(-\infty, \infty)`.
+    c : float, optional
+        Set parameter ``c`` for transformation ``T``. Currently only values
+        between 0 and -0.5 are allowed. If ``c`` is between 0 and -0.5 it is
+        set to -0.5. See Notes for definition of the transformation function
+        ``T``. Default is -0.5.
+    cpoints : int, optional
+        The number of construction points. Default is 30.
+    variant : str, optional
```

Comment:
i think the correct way to document this is 
```
variant : {'ps', 'gw', 'ia'}, optional.
    The variant to use. The default is '...', see Notes for their explanations
```

(note there is a typo in explanation)



https://numpydoc.readthedocs.io/en/latest/format.html#parameters

Reply:
done.